src/mondeath.cpp

```diff
@@ -97,6 +97,31 @@ void mdeath::boomer(monster *z)
     }
 }
 
+void mdeath::boomer_glow(monster *z)
+{
+    std::string explode = string_format(_("a %s explode!"), z->name().c_str());
+    sounds::sound(tripoint(z->posx(), z->posy(), z->posz()), 24, explode);
+    for (int i = -1; i <= 1; i++) {
+        for (int j = -1; j <= 1; j++) {
+            for (int k = -1; k <= 1; k++){
+                g->m.bash( tripoint( z->posx() + i, z->posy() + j, z->posz() + k ), 10 );
+                g->m.add_field(tripoint(z->posx() + i, z->posy() + j, z->posz() + k), fd_bile, 1, 0);
+                int mondex = g->mon_at(tripoint(z->posx() + i, z->posy() + j, z->posz() + k));
```

Comment:
Note that all the tripoints in this block are the exact same tripoint. Declare it at the top of the block and re-use it rather than rebuilding it for every function.
Like this:
```
tripoint dest( x, y, z );
g->m.bash( dest, 10 );
g->m.add_field( dest, fd_bile, 1, 0 );
```
This is much more readable and consistent than re-declaring it every time.

Don't add the z-component yet - the multi-z-level checks don't work yet so the explosion would pass through floors and ceilings. For now just keep the z coord just equal to `z->posz()`.